### DISCUSSION TITLE
WebRTC UDP 포트 설정이 애매합니다. & 사용자 인증서버가 분리되어있을때 통신 방식이 고민됩니다.

### DISCUSSION THREAD

**Discussion by GoByeonghu** [100-hours-a-week/discussions#3] [개발] [answered]
1. WebRTC 미디어 중계 서버가 연결된 피어마다 포트를 할당해야 합니다 (1~60000번 포트). 이 경우, 앞에 웹 서버(Nginx)를 둘 수 있는지 궁금합니다. 만약 둘 수 없다면, Express로 구현된 미디어 서버가 직접 요청을 받아야 하는지 알고 싶습니다.

2. MediaSoup은 HTTPS 연결만 허용하며, HTTP 연결은 거부합니다. 이 경우, Nginx를 사용하면 Nginx에서 복호화되어 HTTP로 전달되는데, 이 문제를 해결할 방법이 있는지 궁금합니다.

3. 사용자 인증 서버가 WebRTC 서버와 분리되어 있습니다. 이 경우, 인증 서버에 인증 요청을 HTTP로 보내야 하는지, 아니면 더 오버헤드가 적은 방식으로 처리할 수 있는지 궁금합니다. 예를 들어, 메시징 방식으로 처리하는 것이 나을지 알고 싶습니다.

**Comment by GoByeonghu** [comment, marked as answer]
1. Nginx는 WAS(Web Application Server) 역할을 수행하기에는 한계가 있습니다. WebRTC가 연결에 할당한 포트를 Nginx도 사용해야 하는데, 이를 동적으로 처리하는 것이 어렵습니다. 따라서 Nginx 설정을 좀 더 조사해보고, 가능하다면 적용해보세요. 만약 불가능하다면, Express 서버에 직접 연결하는 방법을 고려하는 것이 좋습니다.

2. (첫 번째 답변으로 대체)

3. 메시징 시스템과 같이 Queue를 사용하는 방법은 '실시간성'에 문제를 일으킬 수 있습니다. 따라서 HTTP를 통한 직접 연결 방식을 더 추천드립니다.

(작성자: Jeff)

**Reply by ej31** [reply]
![image](https://github.com/user-attachments/assets/7e804731-1f47-4d33-b10b-d67615085a13)
정리 아주 좋았쓰  🎉

**Comment by ej31** [comment]
조오금더 설명 자세히 해주자면..
- 만약 nginx 를 중간에 꼭 끼우고 싶다면 영상을 전송하는 사람은 RTMP + nginx 를 이용해서 업스트림을 구성하고 영상을 시청하는 시청자에게 WebRTC 로 전송하도록 미디어 서버를 구성하면 됩니다.
- 그런데 이렇게 하면 프로젝트를 불도저로 밀어야 하는 상황이니 생각하지 않아도 되는 경우입니다.
